wrappers/core18.go

```diff
@@ -155,6 +195,13 @@ func writeSnapdServicesOnCore(s *snap.Info, inter interacter) error {
 
 	// enable/start all the new services
 	for _, unit := range changed {
+		// enable is not idempotent if the symlink target switches from
+		// /lib to /etc in the multiuser-wantes.d so we need to disable
+		// first (enable --force is not available through the systemd
+		// wrapper)
```

Comment:
could we do `systemctl reenable` (or the Go equivalent, probable needs a new sytemd.Systemd Method) instead? I don't think that is atomic either, but that seems a little bit safer to me since now the race condition window is only during systemd execution, not in between snapd's execution of `systemctl disable` and then `systemctl enable`.

Reply:
Yeah, it's still not atomic: https://github.com/systemd/systemd/blob/master/src/shared/install.c#L2558-L2583